src/sfctl/commands.py

```diff
@@ -267,11 +267,12 @@ def load_command_table(self, args):  # pylint: disable=too-many-statements
                           client_factory=client_create) as group:
             group.command('create', 'create')
             group.command('upgrade', 'upgrade')
+
+        with CommandGroup(self, 'application', 'sfctl.custom_app#{}') as group:
             group.command('upload', 'upload')
 
         # Need an empty client for the select and upload operations
-        with CommandGroup(self, 'cluster', 'sfctl.custom_cluster#{}',
-                          client_factory=client_create) as group:
+        with CommandGroup(self, 'cluster', 'sfctl.custom_cluster#{}') as group:
```

Comment:
will it just pass a null object for the client?

Reply:
I believe it does, and never looks to use the client. This change was made because we never passed the client, but since we passed in the client_factory here, it was expecting it, and breaking the function call